samples/msal-node-samples/on-behalf-of/test/obo-aad.spec.ts

```diff
@@ -0,0 +1,139 @@
+        const envResponse = await labClient.getVarsByCloudEnvironment(labApiParms);
+        [username, accountPwd] = await setupCredentials(envResponse[0], labClient);
+    });
+
+    afterAll(async () => {
+        await browser.close();
+    });
+
+    describe("Acquire Token", () => {
+        let webAppConfidentialClient: ConfidentialClientApplication;
+        let webApiConfidentialClient: ConfidentialClientApplication;
+        let webAppServer: any;
+        let webApiServer: any;
+
+        beforeAll(async () => {
+            webAppConfidentialClient = new ConfidentialClientApplication({auth: webAppConfig.authOptions, cache: { cachePlugin }});
+            webApiConfidentialClient = new ConfidentialClientApplication({auth: webApiConfig.authOptions, cache: { cachePlugin }});
+            webAppServer = acquireTokenByCode(webAppConfidentialClient, webAppConfig.serverPort, webApiConfig.serverPort, webAppConfig.redirectUri, webApiConfig.webApiUrl);
+            webApiServer = acquireTokenObo(webApiConfidentialClient, webApiConfig.serverPort, webApiConfig.authOptions.clientId, webApiConfig.authOptions.authority, webApiConfig.discoveryKeysEndpoint);
+            await NodeCacheTestUtils.resetCache(TEST_CACHE_LOCATION);
+        });
+
+        afterAll(async () => {
+            if (webAppServer) {
+                webAppServer.close();
+            }
+
+            if (webApiServer) {
+                webApiServer.close();
+            }
+        });
+
+        beforeEach(async () => {
+            context = await browser.createIncognitoBrowserContext();
+            page = await context.newPage();
+            page.setDefaultTimeout(5000);
+            page.on("dialog", async dialog => {
+                console.log(dialog.message());
+                await dialog.dismiss();
+            });
+        });
+
+        afterEach(async () => {
+            await page.close();
+            await context.close();
+            await NodeCacheTestUtils.resetCache(TEST_CACHE_LOCATION);
+        });
+
+        it("Performs acquire token via OBO flow", async () => {
+            const screenshot = new Screenshot(`${screenshotFolder}/BaseCase`);
+            await page.goto(HOME_ROUTE);
+            await enterCredentials(page, screenshot, username, accountPwd);
+            await page.waitForFunction(`window.location.href.startsWith("${SAMPLE_HOME_URL}")`);
+
+            const cachedTokens = await NodeCacheTestUtils.waitForTokens(TEST_CACHE_LOCATION, 2000);
+            expect(cachedTokens.accessTokens.length).toBe(2);
+            expect(cachedTokens.idTokens.length).toBe(2);
+            expect(cachedTokens.refreshTokens.length).toBe(2);
+
+            const accounts = await NodeCacheTestUtils.getAccounts(TEST_CACHE_LOCATION);
+            expect(Object.keys(accounts).length).toBe(1);
+            const account = Object.values(accounts)[0];
+            expect(account.username).toEqual(username);
+        });
```

Comment:
Can we add another test case to check if obo retrieves the cached access token (if any) instead of going to the network (essentially covering this [issue](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/5397)and [PR here](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5398))

Reply:
Looks like that test case is already covered here: [lib/msal-common/test/client/OnBehalfOfClient.spec.ts](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/test/client/OnBehalfOfClient.spec.ts#L235)

Reply:
No this is only testing if the token is retrieved from in-memory cache. The issue comes up when restoring the cache from disk to in-memory (its mainly a serialization issue). To test that we would have to restart the server or instantiate a new CCA. But I think we're fine without this -it's only a nice-to-have